src/front/wgsl/mod.rs

```diff
@@ -1813,16 +1815,38 @@ impl Parser {
                 let mut block = Vec::new();
 
                 lexer.expect(Token::Paren('('))?;
-                let initialization =
-                    self.parse_statement(lexer, context.reborrow(), is_uniform_control_flow)?;
-                let condition = self.parse_general_expression(lexer, context.as_expression())?;
-                lexer.expect(Token::Separator(';'))?;
-                let mut continuing = None;
-                // manually parse the next statement here instead of calling parse_statement
-                // because the statement is not terminated with a semicolon
-                if let Token::Word(ident) = lexer.peek() {
+                let initialization = if lexer.skip(Token::Separator(';')) {
+                    None
+                } else {
+                    match self.parse_statement(
+                        lexer,
+                        context.reborrow(),
+                        is_uniform_control_flow,
+                    )? {
+                        statement @ Some(crate::Statement::Store { .. }) => statement,
```

Comment:
it seems to me that the fancy pattern matching here is not necessary.
Instead we could have simpler code by just doing `let statement = self.parse_statement(...); match statement {... }`